apis/openstack-swift/src/main/java/org/jclouds/openstack/swift/v1/blobstore/RegionScopedSwiftBlobStore.java

```diff
@@ -396,6 +401,69 @@ public long countBlobs(String containerName) {
    }
 
    @Override
+   public MultipartUpload initiateMultipartUpload(String container, BlobMetadata blobMetadata) {
+      String uploadId = UUID.randomUUID().toString();
+      return MultipartUpload.create(container, blobMetadata.getName(), uploadId, blobMetadata);
+   }
+
+   @Override
+   public void abortMultipartUpload(MultipartUpload mpu) {
+      ImmutableList.Builder<String> names = ImmutableList.builder();
+      for (MultipartPart part : listMultipartUpload(mpu)) {
+         names.add(mpu.blobName() + "-" + part.partNumber());
+      }
+      removeBlobs(mpu.containerName(), names.build());
+   }
+
+   @Override
+   public String completeMultipartUpload(MultipartUpload mpu, List<MultipartPart> parts) {
+      ImmutableList.Builder<Segment> builder = ImmutableList.builder();
+      for (MultipartPart part : parts) {
+         String path = mpu.containerName() + "/" + mpu.blobName() + "-" + part.partNumber();
+         builder.add(Segment.builder().path(path).etag(part.partETag()).sizeBytes(part.partSize()).build());
+      }
+      Map<String, String> metadata = ImmutableMap.of();  // TODO: how to populate this from mpu.blobMetadata()?
+      return api.getStaticLargeObjectApi(regionId, mpu.containerName()).replaceManifest(mpu.blobName(),
+            builder.build(), metadata);
+   }
+
+   @Override
+   public MultipartPart uploadMultipartPart(MultipartUpload mpu, int partNumber, Payload payload) {
+      String partName = mpu.blobName() + "-" + partNumber;
+      String eTag = api.getObjectApi(regionId, mpu.containerName()).put(partName, payload);
+      // TODO: only works for payloads where the caller specified the length.  How else can we get at the size?
+      // ObjectApi.getWithoutBody does not return the size.
```

Comment:
Actually content metadata returns the content-length without having to get the whole object.
http://docs.rackspace.com/files/api/v1/cf-devguide/content/HEAD_retrieveobjectmeta_v1__account___container___object__objectServicesOperations_d1e000.html

However, at the moment it gets filtered out I think. So this might require additional work first.